### ISSUE TITLE
Are the predictions printed shuffled or in serial according to test dataset in  run_race.py?

### ISSUE THREAD

**Issue by Sumeghtech** [zihangdai/xlnet#92]
I was able to print the `eval_input_dict` in `run_race.py` for generating predictions.

Now, how do I understand the order of the predictions printed? Like the predictions aren't serially printed according to the file reading system.
@kimiyoung .

Example: 
I0630 19:27:21.468778 139938823763840 tpu_estimator.py:590] Enqueue next (1) batch(es) of data to infeed.
I0630 19:27:21.469814 139938823763840 tpu_estimator.py:594] Dequeue next (1) batch(es) of data from outfeed.
I0630 19:27:21.471143 139937746306816 tpu_estimator.py:275] Outfeed finished for iteration (0, 0)
{'labels': 1, 'predictions': 3, 'logits': array([-0.12642628, -0.0991993 , -0.13819037, -0.0714176 ], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 2, 'logits': array([-0.25851762, -0.14858288, -0.08244841, -0.15488403], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 0, 'logits': array([ 0.02535003, -0.1628823 , -0.49583822, -0.4239014 ], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 1, 'logits': array([-0.12479863,  0.1504738 ,  0.02581257,  0.04662573], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 1, 'logits': array([-0.09238349, -0.04598972, -0.07070735, -0.12756059], dtype=float32), 'weights': 1.0}
{'labels': 2, 'predictions': 1, 'logits': array([-0.12795207, -0.02986489, -0.23880608, -0.20500727], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 1, 'logits': array([-0.02972522,  0.0425356 ,  0.03581315, -0.02963182], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 1, 'logits': array([-0.00578013,  0.00338969, -0.03053465, -0.01934794], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 0, 'logits': array([ 0.27454737,  0.16609548, -0.20225091, -0.3271135 ], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 0, 'logits': array([ 0.1587551 , -0.13514623, -0.03806397, -0.47514313], dtype=float32), 'weights': 1.0}
{'labels': 2, 'predictions': 1, 'logits': array([ 0.10922289,  0.12710331,  0.02629827, -0.0787103 ], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 0, 'logits': array([0.16017492, 0.02594006, 0.09249569, 0.09899483], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 0, 'logits': array([-0.18771921, -0.23714969, -0.22389576, -0.23356643], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 0, 'logits': array([ 0.02847612, -0.01706637, -0.14055501, -0.05842215], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 1, 'logits': array([-0.03821297,  0.02404073, -0.08221474, -0.05104077], dtype=float32), 'weights': 1.0}
{'labels': 2, 'predictions': 1, 'logits': array([-0.07974827,  0.11125989,  0.00440803,  0.0060682 ], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 2, 'logits': array([-0.25667855, -0.23453096, -0.18907604, -0.2183947 ], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 3, 'logits': array([-0.11256263, -0.02653629, -0.11111617,  0.02528933], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 3, 'logits': array([-0.06740214, -0.1597581 , -0.11641204, -0.02050325], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 1, 'logits': array([-0.26551592, -0.12319008, -0.21057466, -0.27629927], dtype=float32), 'weights': 1.0}
{'labels': 2, 'predictions': 0, 'logits': array([-0.3514495 , -0.38794804, -0.40034616, -0.404324  ], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 0, 'logits': array([-0.24498221, -0.35472882, -0.27922606, -0.24510363], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 2, 'logits': array([-0.2819972 , -0.01811412, -0.00702809, -0.08678977], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 3, 'logits': array([-0.11946625, -0.21603471, -0.3624884 , -0.09108685], dtype=float32), 'weights': 1.0}
{'labels': 2, 'predictions': 2, 'logits': array([-0.11633293, -0.16468103, -0.09840389, -0.18640852], dtype=float32), 'weights': 1.0}
{'labels': 0, 'predictions': 2, 'logits': array([-0.22886163, -0.10918848,  0.01751065, -0.15580115], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 3, 'logits': array([-0.29510933, -0.32492617, -0.3219505 , -0.19274434], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 0, 'logits': array([-0.11171978, -0.23901683, -0.13906357, -0.18748996], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 0, 'logits': array([-0.0248003 , -0.12337333, -0.18529966, -0.05989137], dtype=float32), 'weights': 1.0}
{'labels': 3, 'predictions': 3, 'logits': array([-0.1324261 , -0.22223294, -0.25643134, -0.03107378], dtype=float32), 'weights': 1.0}
{'labels': 2, 'predictions': 3, 'logits': array([-0.03233376, -0.18092477, -0.1726094 , -0.02796167], dtype=float32), 'weights': 1.0}
{'labels': 1, 'predictions': 3, 'logits': array([-0.32592696, -0.34190968, -0.31606865, -0.0957111 ], dtype=float32), 'weights': 1.0}

**Comment by Sumeghtech** [comment]
Any help @kimiyoung @zihangdai ?

**Comment by kimiyoung** [comment]
It's in the same order.
